### PR TITLE
impl drag items from nei

### PR DESCRIPTION
**Pull request by GlodBlock**
allow item drag from nei.

### CODE REVIEW

**Comment by GTNHAfx237v7** [comment]
Not sure what the update here is - item drag from NEI has always been possible?

**Comment by GlodBlock** [comment]
item drag is added by nee, now put it in ae2

**Comment by repo-alt** [comment]
How will it coexist with NEE implementation?

File: src/main/java/appeng/util/Pair.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
+package appeng.util;
```
Comment (repo-alt) [inline src/main/java/appeng/util/Pair.java:1]: There is javafx.util.Pair which is a part of JRE/JDK 8

**Comment by GlodBlock** [comment]
> How will it coexist with NEE implementation?

can coexist

File: src/main/java/appeng/util/Pair.java (hunk 1 above)
Comment (GlodBlock) [reply src/main/java/appeng/util/Pair.java:1]: my dev running environment crashed when using javafx.util.Pair
```
Caused by: java.lang.NoClassDefFoundError: javafx/util/Pair

	at codechicken.nei.ItemPanel.handleClickExt(ItemPanel.java:323) ~[ItemPanel.class:?]
	at codechicken.nei.LayoutManager.mouseClicked(LayoutManager.java:171) ~[LayoutManager.class:?]
	at codechicken.nei.guihook.GuiContainerManager.mouseClicked(GuiContainerManager.java:368) ~[GuiContainerManager.class:?]
	at net.minecraft.client.gui.inventory.GuiContainer.mouseClicked(GuiContainer.java) ~[GuiContainer.class:?]
	at appeng.client.gui.AEBaseGui.mouseClicked(AEBaseGui.java:340) ~[AEBaseGui.class:?]
	at net.minecraft.client.gui.GuiScreen.handleMouseInput(GuiScreen.java:296) ~[GuiScreen.class:?]
	at net.minecraft.client.gui.inventory.GuiContainer.handleMouseInput(GuiContainer.java) ~[GuiContainer.class:?]
	at appeng.client.gui.AEBaseGui.handleMouseInput(AEBaseGui.java:715) ~[AEBaseGui.class:?]
	at net.minecraft.client.gui.GuiScreen.handleInput(GuiScreen.java:268) ~[GuiScreen.class:?]
	at net.minecraft.client.Minecraft.runTick(Minecraft.java:1640) ~[Minecraft.class:?]
	... 11 more
```

Comment (repo-alt) [reply src/main/java/appeng/util/Pair.java:1]: weird, looks like JavaFX is not part of newer JDKs (version > 9)

Comment (slprime) [reply src/main/java/appeng/util/Pair.java:1]: maybe ```import akka.japi.Pair;``` ?

Comment (repo-alt) [reply src/main/java/appeng/util/Pair.java:1]: Is it "more standard" than JavaFX? or it is used my minecraft itself?

Comment (GlodBlock) [reply src/main/java/appeng/util/Pair.java:1]: akka's pair works in my dev and real game environment. but bartworks' pair is also written by itself so i am not sure whether we should use it.

Comment (repo-alt) [reply src/main/java/appeng/util/Pair.java:1]: We should probably remove it too?

Comment (GlodBlock) [reply src/main/java/appeng/util/Pair.java:1]: then let's try to use akka'a pair. at least it doesn't crash or behave weird in my machine

**Review by repo-alt — APPROVED** [review]

**Comment by vfyjxf** [comment]
> How will it coexist with NEE implementation?

NEE implementation can be disabled in config file.

**Pull request merged**